src/coreclr/jit/lclmorph.cpp

```diff
@@ -1103,6 +1103,75 @@ class LocalAddressVisitor final : public GenTreeVisitor<LocalAddressVisitor>
                 break;
             }
 
+            case GT_INDEX_ADDR:
+            {
+                assert(TopValue(2).Node() == node);
+                assert(TopValue(1).Node() == node->gtGetOp1());
+                assert(TopValue(0).Node() == node->gtGetOp2());
+
+                if (node->gtGetOp2()->IsCnsIntOrI() && TopValue(1).IsAddress())
```

Comment:
Add a comment here that we only expect `TopValue(1).IsAddress()` to be true for stack allocated arrays?